jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -635,6 +635,35 @@ protected void doStart() throws Exception
         }
     }
 
+    private Server.Shutdown getShutdown()
+    {
+        Server.Shutdown shutdown = getBean(Shutdown.class);
+        if (shutdown != null)
+            return shutdown;
+
+        // Look for the old-school (now deprecated) ShutdownMonitor
+        // First we look to see if anyone has called ShutdownMonitor.getInstance() yet.
+        //noinspection removal
+        ShutdownMonitor shutdownMonitor = ShutdownMonitor.INSTANCE.get();
+        //noinspection removal
+        if (shutdownMonitor != null && shutdownMonitor.getPort() != (-1))
+        {
+            addBean(shutdownMonitor);
+            return shutdownMonitor;
+        }
+
+        // Lets create the default ServerShutdown implementation.
+        // Start System Property version
+        ServerShutdown serverShutdown = new ServerShutdown();
+        if (serverShutdown.isConfigurationValid())
```

Comment:
This is really checking whether or not the user wants to use the shutdown service - indicated by having configured values for the port. So it's really checking `isEnabled()`, not `isConfigurationValid()`.